Sources/Session.php

```diff
@@ -95,151 +98,157 @@ function loadSession()
-}
-
-/**
- * Implementation of sessionRead() replacing the standard read handler.
- *
- * @param string $session_id The session ID
- * @return string The session data
- */
-function sessionRead($session_id)
-{
-	global $smcFunc;
-
-	if (preg_match('~^[A-Za-z0-9,-]{16,64}$~', $session_id) == 0)
-		return '';
-
-	// Look for it in the database.
-	$result = $smcFunc['db_query']('', '
-		SELECT data
-		FROM {db_prefix}sessions
-		WHERE session_id = {string:session_id}
-		LIMIT 1',
-		array(
-			'session_id' => $session_id,
-		)
-	);
-	list ($sess_data) = $smcFunc['db_fetch_row']($result);
-	$smcFunc['db_free_result']($result);
-
-	return $sess_data != null ? $sess_data : '';
-}
-
-/**
- * Implementation of sessionWrite() replacing the standard write handler.
- *
- * @param string $session_id The session ID
- * @param string $data The data to write to the session
- * @return boolean Whether the info was successfully written
- */
-function sessionWrite($session_id, $data)
-{
-	global $smcFunc, $db_connection, $db_server, $db_name, $db_user, $db_passwd;
-	global $db_prefix, $db_persist, $db_port, $db_mb4;
+	/**
+	 * Implementation of SessionHandler::open() replacing the standard open handler.
+	 * It simply returns true.
+	 *
+	 * @param string $path The path to save the session to
+	 * @param string $name The name of the session
+	 * @return boolean Always returns true
+	 */
+	function open(/*PHP 8.0 string*/$path, /*PHP 8.0 string*/$name): bool
+	{
+		return true;
+	}
 
-	if (preg_match('~^[A-Za-z0-9,-]{16,64}$~', $session_id) == 0)
-		return false;
+	/**
+	 * Implementation of SessionHandler::close() replacing the standard close handler.
+	 * It simply returns true.
+	 *
+	 * @return boolean Always returns true
+	 */
+	public function close(): bool
+	{
+		return true;
+	}
 
-	// php < 7.0 need this
-	if (empty($db_connection))
+	/**
+	 * Implementation of SessionHandler::read() replacing the standard read handler.
+	 *
+	 * @param string $id The session ID
+	 * @return string The session data
+	 */
+	#[\ReturnTypeWillChange]
+	public function read(/*PHP 8.0 string*/$id)/*PHP 8.0: string|false*/
 	{
-		$db_options = array();
+		global $smcFunc;
+
+		if (!$this->IsValidSessionID($id))
+			return '';
+
+		// Look for it in the database.
+		$result = $smcFunc['db_query']('', '
+			SELECT data
+			FROM {db_prefix}sessions
+			WHERE session_id = {string:session_id}
+			LIMIT 1',
+			array(
+				'session_id' => $id,
+			)
+		);
+		list ($sess_data) = $smcFunc['db_fetch_row']($result);
+		$smcFunc['db_free_result']($result);
+
+		return $sess_data != null ? $sess_data : '';
+	}
 
-		// Add in the port if needed
-		if (!empty($db_port))
-			$db_options['port'] = $db_port;
+	/**
+	 * Implementation of SessionHandler::write() replacing the standard write handler.
+	 *
+	 * @param string $id The session ID
+	 * @param string $data The data to write to the session
+	 * @return boolean Whether the info was successfully written
+	 */
+	#[\ReturnTypeWillChange]
```

Comment:
Weren't attributes only introduced in PHP 8.2? Maybe do a version check to keep using the old code in older php versions, similar to the db layer in 2.0.14.

Reply:
Correct, but # is also supported for comments.  So for anything below 8.2, its ignored as a comment.  For 8.2, this is telling PHP to ignore the fact the signature doesn't match. 8.1 and 8.0 don't seem to care the signature doesn't match the implantation interfaces.  Thus left 7.x, which complained about the signatures not matching since it was type hinted and the interface in 7.x was without type hints.